lib/pbench/server/api/resources/query_apis/metadata_index/iteration_samples.py

```diff
@@ -0,0 +1,347 @@
+               "sample.server_hostname":[
+                  "127.0.0.1",
+                  "all"
+               ],
+               "sample.uid":[
+                  "client_hostname:127.0.0.1-server_hostname:127.0.0.1-server_port:20010",
+                  "client_hostname:all-server_hostname:all-server_port:all"
+               ],
+               "sample.uid_tmpl":[
+                  "client_hostname:%client_hostname%-server_hostname:%server_hostname%-server_port:%server_port%"
+               ]
+            }
+        """
+        try:
+            new_json = {
+                key: [bucket["key"] for bucket in agg["buckets"]]
+                for key, agg in es_json["aggregations"].items()
+                if agg["buckets"]
+            }
+            return jsonify(new_json)
+        except KeyError as e:
+            raise PostprocessError(
+                HTTPStatus.INTERNAL_SERVER_ERROR,
+                f"Can't find Elasticsearch match data '{e}' in {es_json!r}",
+            )
+
+
+class IterationSamplesRows(RunIdBase):
+    """
+    Iteration samples API that returns iteration sample rows after
+    applying client specified filters.
+    """
+
+    DOCUMENT_SIZE = 10000  # Number of documents to return in one page
+
+    def __init__(self, config: PbenchServerConfig, logger: Logger):
+        super().__init__(
+            config,
+            logger,
+            Schema(
+                Parameter("filters", ParamType.JSON, required=False),
+                Parameter("run_id", ParamType.STRING, required=True),
+                Parameter("scroll_id", ParamType.STRING, required=False),
+            ),
+        )
+
+    def assemble(self, json_data: JSON, context: CONTEXT) -> JSON:
+        """
+        Construct a pbench Elasticsearch query for filtering iteration
+        samples based on a given run id and other filtering parameters
+        mention in JSON payload.
+
+        Note: If the ES scroll id is present we will ignore the filters
+        parameter and instead construct an Elasticsearch query for scrolling
+        based on a client provided scroll id.
+
+        If a scroll_id is specified, the query will return the next page of
+        results of the original query; otherwise we form a new query, either
+        with the specified filters or without any filters (returning up to 10000
+        documents at a time).
+
+        Args:
+            json_data:
+            "run_id": run id of a dataset that client wants to search for
+                    fetching the iteration samples.
+
+            "scroll_id": Server provided Elasticsearch scroll id that client
+                        recieved in the result of the original query with
+                        filters. This can be used to fetch the next page of
+                        the result.
+
+            "filters": key-value representation of query filter parameters to
+                       narrow the search results e.g. {"sample.name": "sample1"}
```

Comment:
These should be indented a tab stop; and, ideally, the continuation line indentation would match down the list.